pymc3/tests/test_distributions.py

```diff
@@ -889,14 +889,18 @@ def test_half_flat(self):
         assert 0.0 == HalfFlat.dist().logcdf(np.inf).tag.test_value
         assert -np.inf == HalfFlat.dist().logcdf(-np.inf).tag.test_value
 
-    def test_normal(self):
+    def test_normal_logp(self):
         self.check_logp(
             Normal,
             R,
             {"mu": R, "sigma": Rplus},
             lambda value, mu, sigma: sp.norm.logpdf(value, mu, sigma),
             decimal=select_by_precision(float64=6, float32=1),
         )
+
+    @pytest.mark.xfail(condition=(aesara.config.floatX == "float32"), reason="Fails on float32")
+    # this test doesn't always result in the same outcome (pass/not pass)
```

Comment:
How did you test this in float32? Locally or in here?

To make it reproducible you can temporarily set `n_samples=-1` in `check_logp`/`check_logcdf`. This will force it to run all permutations of the test instead of a limited subset.

Just want to confirm this is not an issue we have overlooked since the test was not marked as xfail before. Maybe it will be enough to lower the float32 precision like is being done for check_logp already.

Reply:
I ran all the tests locally by setting floatX to float32

After having set `n_samples=-1` it returns the same result (failure) consistently, thanks. I am going to remove the comment

Reply:
This one should be scrutinized a bit more because it was not marked as xfail before (right?). 

Do you feel capable of identifying which values are failing (input -> output -> expected_output in float32)? You might need to temporarily set a try/except in check_logcdf to not interrupt the test prematurely. Let me know if you need help with this.

Reply:
We can wait and see if it comes as an xpass here to be sure there is actually a problem, before delving deeper.

Reply:
Indeed it XFAILED: https://github.com/pymc-devs/pymc3/pull/4497/checks?check_run_id=2021551266

```pymc3/tests/test_distributions.py::TestMatchesScipy::test_normal_logcdf XFAIL [  2%]```